lib/MaxMind/DB/Writer/Tree.pm

```diff
@@ -287,12 +331,12 @@ sub new_from_frozen_tree {
 
     $params->{database_type} = $database_type if defined $database_type;
     $params->{description}   = $description   if defined $description;
+    $params->{merge_strategy} = $merge_strategy if defined $merge_strategy;
 
     my $tree = _thaw_tree(
         $filename,
         $params_size + 4,
-        map { $params->{$_} }
-            qw( ip_version record_size merge_record_collisions ),
+        map { $params->{$_} } qw( ip_version record_size merge_strategy ),
```

Comment:
So, this code introduces a subtle backward compatibility for stuff frozen with the old version of the code and unfrozen with the new in the situation where merge_record_collision was set to `1` in the frozen tree and you set merge_strategy to `none`.

Do we need to worry about this given frozen trees aren't considered compatible?  We should either address the problem or put a note in the release notes